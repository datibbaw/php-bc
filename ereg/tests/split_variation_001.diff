006+ Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
006- Error: 8192 - Function split() is deprecated, %s(74)
013+ Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
013- Error: 8192 - Function split() is deprecated, %s(74)
026+ Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
026- Error: 8192 - Function split() is deprecated, %s(74)
033+ Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
033- Error: 8192 - Function split() is deprecated, %s(74)
040+ Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
040- Error: 8192 - Function split() is deprecated, %s(74)
047+ Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
047- Error: 8192 - Function split() is deprecated, %s(74)
054+ Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
054- Error: 8192 - Function split() is deprecated, %s(74)
061+ Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
061- Error: 8192 - Function split() is deprecated, %s(74)
068+ Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
068- Error: 8192 - Function split() is deprecated, %s(74)
076+ Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
077+ Error: 2 - addcslashes() expects parameter 1 to be string, array given, /Users/tjerk/Documents/php-bc/ereg/ereg.php(49)
078+ array(5) {
079+   [0]=>
080+   string(0) ""
081+   [1]=>
082+   string(1) "1"
083+   [2]=>
084+   string(1) " "
085+   [3]=>
086+   string(1) "a"
087+   [4]=>
088+   string(13) " 1 Array 1 c "
089+ }
076- Error: 8192 - Function split() is deprecated, %s(74)
077- Error: 2 - split() expects parameter 1 to be string, array given, %s(74)
078- NULL
079- Error: 8 - Array to string conversion, %ssplit_variation_001.php(%d)
080- 
081- Arg value Array 
082- Error: 8192 - Function split() is deprecated, %s(74)
083- Error: 2 - split() expects parameter 1 to be string, array given, %s(74)
084- NULL
088- Error: 8192 - Function split() is deprecated, %s(74)
089- Error: 2 - split() expects parameter 1 to be string, array given, %s(74)
090- NULL
091- Error: 8 - Array to string conversion, %ssplit_variation_001.php(%d)
092- 
093- Arg value Array 
094- Error: 8192 - Function split() is deprecated, %s(74)
095- Error: 2 - split() expects parameter 1 to be string, array given, %s(74)
096- NULL
093+ Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
094+ Error: 2 - addcslashes() expects parameter 1 to be string, array given, /Users/tjerk/Documents/php-bc/ereg/ereg.php(49)
095+ array(5) {
096+   [0]=>
097+   string(0) ""
098+   [1]=>
099+   string(1) "1"
100+   [2]=>
101+   string(1) " "
102+   [3]=>
103+   string(1) "a"
104+   [4]=>
105+   string(13) " 1 Array 1 c "
106+ }
100- Error: 8192 - Function split() is deprecated, %s(74)
101- Error: 2 - split() expects parameter 1 to be string, array given, %s(74)
102- NULL
103- 
104- Arg value  
105- Error: 8192 - Function split() is deprecated, %s(74)
106- Error: 2 - split(): REG_EMPTY, %s(74)
107- bool(false)
109- Arg value  
110- Error: 8192 - Function split() is deprecated, %s(74)
111- Error: 2 - split(): REG_EMPTY, %s(74)
112- bool(false)
113- 
114- Arg value 1 
115- Error: 8192 - Function split() is deprecated, %s(74)
116- array(4) {
110+ Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
111+ Error: 2 - addcslashes() expects parameter 1 to be string, array given, /Users/tjerk/Documents/php-bc/ereg/ereg.php(49)
112+ array(5) {
113+   [0]=>
114+   string(0) ""
115+   [1]=>
116+   string(1) "1"
117+   [2]=>
118+   string(1) " "
119+   [3]=>
120+   string(1) "a"
121+   [4]=>
122+   string(13) " 1 Array 1 c "
123+ }
124+ Error: 8 - Array to string conversion, /Users/tjerk/Documents/php-bc/ereg/tests/split_variation_001.php(73)
120-   string(3) " a "
122-   string(7) " Array "
124-   string(3) " c "
126+ Arg value Array 
127+ Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
128+ Error: 2 - addcslashes() expects parameter 1 to be string, array given, /Users/tjerk/Documents/php-bc/ereg/ereg.php(49)
129+ array(5) {
127- Arg value  
128- Error: 8192 - Function split() is deprecated, %s(74)
129- Error: 2 - split(): REG_EMPTY, %s(74)
130- bool(false)
131- 
132- Arg value 1 
133- Error: 8192 - Function split() is deprecated, %s(74)
134- array(4) {
133+   string(1) "1"
135+   string(1) " "
137+   string(1) "a"
138+   [4]=>
139+   string(13) " 1 Array 1 c "
138-   string(3) " a "
140-   string(7) " Array "
141+ Error: 8 - Array to string conversion, /Users/tjerk/Documents/php-bc/ereg/tests/split_variation_001.php(73)
142-   string(3) " c "
143+ Arg value Array 
144+ Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
145+ Error: 2 - addcslashes() expects parameter 1 to be string, array given, /Users/tjerk/Documents/php-bc/ereg/ereg.php(49)
146+ array(5) {
146- Error: 8192 - Function split() is deprecated, %s(74)
147- Error: 2 - split(): REG_EMPTY, %s(74)
148- bool(false)
150+   string(1) "1"
151- Error: 8192 - Function split() is deprecated, %s(74)
152- Error: 2 - split(): REG_EMPTY, %s(74)
153- bool(false)
152+   string(1) " "
154+   string(1) "a"
155+   [4]=>
156+   string(13) " 1 Array 1 c "
155- Arg value  
156- Error: 8192 - Function split() is deprecated, %s(74)
157- Error: 2 - split(): REG_EMPTY, %s(74)
158- bool(false)
159- Error: 4096 - Object of class stdClass could not be converted to string, %s(73)
160+ Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
161+ array(5) {
162+   [0]=>
163+   string(0) ""
164+   [1]=>
165+   string(1) "1"
166+   [2]=>
167+   string(1) " "
168+   [3]=>
169+   string(1) "a"
170+   [4]=>
171+   string(13) " 1 Array 1 c "
172+ }
162- Error: 8192 - Function split() is deprecated, %s(74)
163- Error: 2 - split() expects parameter 1 to be string, object given, %s(74)
164- NULL
166- Arg value  
167- Error: 8192 - Function split() is deprecated, %s(74)
168- Error: 2 - split(): REG_EMPTY, %s(74)
169- bool(false)
172- Error: 8192 - Function split() is deprecated, %s(74)
173- Error: 2 - split(): REG_EMPTY, %s(74)
174- bool(false)
175- Done
175+ Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
176+ array(5) {
177+   [0]=>
178+   string(0) ""
179+   [1]=>
180+   string(1) "1"
181+   [2]=>
182+   string(1) " "
183+   [3]=>
184+   string(1) "a"
185+   [4]=>
186+   string(13) " 1 Array 1 c "
187+ }
189+ Arg value 1 
190+ Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
191+ array(4) {
192+   [0]=>
193+   string(0) ""
194+   [1]=>
195+   string(3) " a "
196+   [2]=>
197+   string(7) " Array "
198+   [3]=>
199+   string(3) " c "
200+ }
203+ Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
204+ array(5) {
205+   [0]=>
206+   string(0) ""
207+   [1]=>
208+   string(1) "1"
209+   [2]=>
210+   string(1) " "
211+   [3]=>
212+   string(1) "a"
213+   [4]=>
214+   string(13) " 1 Array 1 c "
215+ }
217+ Arg value 1 
218+ Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
219+ array(4) {
220+   [0]=>
221+   string(0) ""
222+   [1]=>
223+   string(3) " a "
224+   [2]=>
225+   string(7) " Array "
226+   [3]=>
227+   string(3) " c "
228+ }
231+ Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
232+ array(5) {
233+   [0]=>
234+   string(0) ""
235+   [1]=>
236+   string(1) "1"
237+   [2]=>
238+   string(1) " "
239+   [3]=>
240+   string(1) "a"
241+   [4]=>
242+   string(13) " 1 Array 1 c "
243+ }
244+ 
245+ Arg value  
246+ Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
247+ array(5) {
248+   [0]=>
249+   string(0) ""
250+   [1]=>
251+   string(1) "1"
252+   [2]=>
253+   string(1) " "
254+   [3]=>
255+   string(1) "a"
256+   [4]=>
257+   string(13) " 1 Array 1 c "
258+ }
259+ 
260+ Arg value  
261+ Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
262+ array(5) {
263+   [0]=>
264+   string(0) ""
265+   [1]=>
266+   string(1) "1"
267+   [2]=>
268+   string(1) " "
269+   [3]=>
270+   string(1) "a"
271+   [4]=>
272+   string(13) " 1 Array 1 c "
273+ }
274+ Error: 4096 - Object of class stdClass could not be converted to string, /Users/tjerk/Documents/php-bc/ereg/tests/split_variation_001.php(73)
275+ 
276+ Arg value  
277+ Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
278+ Error: 4096 - Object of class stdClass could not be converted to string, /Users/tjerk/Documents/php-bc/ereg/ereg.php(46)
279+ array(5) {
280+   [0]=>
281+   string(0) ""
282+   [1]=>
283+   string(1) "1"
284+   [2]=>
285+   string(1) " "
286+   [3]=>
287+   string(1) "a"
288+   [4]=>
289+   string(13) " 1 Array 1 c "
290+ }
291+ 
292+ Arg value  
293+ Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
294+ array(5) {
295+   [0]=>
296+   string(0) ""
297+   [1]=>
298+   string(1) "1"
299+   [2]=>
300+   string(1) " "
301+   [3]=>
302+   string(1) "a"
303+   [4]=>
304+   string(13) " 1 Array 1 c "
305+ }
306+ 
307+ Arg value  
308+ Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
309+ array(5) {
310+   [0]=>
311+   string(0) ""
312+   [1]=>
313+   string(1) "1"
314+   [2]=>
315+   string(1) " "
316+   [3]=>
317+   string(1) "a"
318+   [4]=>
319+   string(13) " 1 Array 1 c "
320+ }
321+ Done