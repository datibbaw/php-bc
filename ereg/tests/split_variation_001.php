<?php
/* Prototype  : proto array split(string pattern, string string [, int limit])
 * Description: Split string into array by regular expression 
 * Source code: ext/standard/reg.c
 * Alias to functions: 
 */

function test_error_handler($err_no, $err_msg, $filename, $linenum, $vars) {
	echo "Error: $err_no - $err_msg, $filename($linenum)\n";
}
set_error_handler('test_error_handler');

echo "*** Testing split() : usage variations ***\n";

// Initialise function arguments not being substituted (if any)
$string = '1 a 1 Array 1 c ';
$limit = 5;

//get an unset variable
$unset_var = 10;
unset ($unset_var);

//array of values to iterate over
$values = array(

      // int data
      0,
      1,
      12345,
      -2345,

      // float data
      10.5,
      -10.5,
      10.1234567e10,
      10.7654321E-10,
      .5,

      // array data
      array(),
      array(0),
      array(1),
      array(1, 2),
      array('color' => 'red', 'item' => 'pen'),

      // null data
      NULL,
      null,

      // boolean data
      true,
      false,
      TRUE,
      FALSE,

      // empty data
      "",
      '',

      // object data
      new stdclass(),

      // undefined data
      $undefined_var,

      // unset data
      $unset_var,
);

// loop through each element of the array for pattern

foreach($values as $value) {
      echo "\nArg value $value \n";
      var_dump( split($value, $string, $limit) );
};

echo "Done";
?>
