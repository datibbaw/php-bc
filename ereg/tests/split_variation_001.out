*** Testing split() : usage variations ***
Error: 8 - Undefined variable: undefined_var, /Users/tjerk/Documents/php-bc/ereg/tests/split_variation_001.php(64)
Error: 8 - Undefined variable: unset_var, /Users/tjerk/Documents/php-bc/ereg/tests/split_variation_001.php(67)

Arg value 0 
Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
array(1) {
  [0]=>
  string(16) "1 a 1 Array 1 c "
}

Arg value 1 
Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
array(4) {
  [0]=>
  string(0) ""
  [1]=>
  string(3) " a "
  [2]=>
  string(7) " Array "
  [3]=>
  string(3) " c "
}

Arg value 12345 
Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
array(1) {
  [0]=>
  string(16) "1 a 1 Array 1 c "
}

Arg value -2345 
Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
array(1) {
  [0]=>
  string(16) "1 a 1 Array 1 c "
}

Arg value 10.5 
Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
array(1) {
  [0]=>
  string(16) "1 a 1 Array 1 c "
}

Arg value -10.5 
Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
array(1) {
  [0]=>
  string(16) "1 a 1 Array 1 c "
}

Arg value 101234567000 
Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
array(1) {
  [0]=>
  string(16) "1 a 1 Array 1 c "
}

Arg value 1.07654321E-9 
Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
array(1) {
  [0]=>
  string(16) "1 a 1 Array 1 c "
}

Arg value 0.5 
Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
array(1) {
  [0]=>
  string(16) "1 a 1 Array 1 c "
}
Error: 8 - Array to string conversion, /Users/tjerk/Documents/php-bc/ereg/tests/split_variation_001.php(73)

Arg value Array 
Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
Error: 2 - addcslashes() expects parameter 1 to be string, array given, /Users/tjerk/Documents/php-bc/ereg/ereg.php(49)
array(5) {
  [0]=>
  string(0) ""
  [1]=>
  string(1) "1"
  [2]=>
  string(1) " "
  [3]=>
  string(1) "a"
  [4]=>
  string(13) " 1 Array 1 c "
}
Error: 8 - Array to string conversion, /Users/tjerk/Documents/php-bc/ereg/tests/split_variation_001.php(73)

Arg value Array 
Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
Error: 2 - addcslashes() expects parameter 1 to be string, array given, /Users/tjerk/Documents/php-bc/ereg/ereg.php(49)
array(5) {
  [0]=>
  string(0) ""
  [1]=>
  string(1) "1"
  [2]=>
  string(1) " "
  [3]=>
  string(1) "a"
  [4]=>
  string(13) " 1 Array 1 c "
}
Error: 8 - Array to string conversion, /Users/tjerk/Documents/php-bc/ereg/tests/split_variation_001.php(73)

Arg value Array 
Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
Error: 2 - addcslashes() expects parameter 1 to be string, array given, /Users/tjerk/Documents/php-bc/ereg/ereg.php(49)
array(5) {
  [0]=>
  string(0) ""
  [1]=>
  string(1) "1"
  [2]=>
  string(1) " "
  [3]=>
  string(1) "a"
  [4]=>
  string(13) " 1 Array 1 c "
}
Error: 8 - Array to string conversion, /Users/tjerk/Documents/php-bc/ereg/tests/split_variation_001.php(73)

Arg value Array 
Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
Error: 2 - addcslashes() expects parameter 1 to be string, array given, /Users/tjerk/Documents/php-bc/ereg/ereg.php(49)
array(5) {
  [0]=>
  string(0) ""
  [1]=>
  string(1) "1"
  [2]=>
  string(1) " "
  [3]=>
  string(1) "a"
  [4]=>
  string(13) " 1 Array 1 c "
}
Error: 8 - Array to string conversion, /Users/tjerk/Documents/php-bc/ereg/tests/split_variation_001.php(73)

Arg value Array 
Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
Error: 2 - addcslashes() expects parameter 1 to be string, array given, /Users/tjerk/Documents/php-bc/ereg/ereg.php(49)
array(5) {
  [0]=>
  string(0) ""
  [1]=>
  string(1) "1"
  [2]=>
  string(1) " "
  [3]=>
  string(1) "a"
  [4]=>
  string(13) " 1 Array 1 c "
}

Arg value  
Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
array(5) {
  [0]=>
  string(0) ""
  [1]=>
  string(1) "1"
  [2]=>
  string(1) " "
  [3]=>
  string(1) "a"
  [4]=>
  string(13) " 1 Array 1 c "
}

Arg value  
Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
array(5) {
  [0]=>
  string(0) ""
  [1]=>
  string(1) "1"
  [2]=>
  string(1) " "
  [3]=>
  string(1) "a"
  [4]=>
  string(13) " 1 Array 1 c "
}

Arg value 1 
Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
array(4) {
  [0]=>
  string(0) ""
  [1]=>
  string(3) " a "
  [2]=>
  string(7) " Array "
  [3]=>
  string(3) " c "
}

Arg value  
Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
array(5) {
  [0]=>
  string(0) ""
  [1]=>
  string(1) "1"
  [2]=>
  string(1) " "
  [3]=>
  string(1) "a"
  [4]=>
  string(13) " 1 Array 1 c "
}

Arg value 1 
Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
array(4) {
  [0]=>
  string(0) ""
  [1]=>
  string(3) " a "
  [2]=>
  string(7) " Array "
  [3]=>
  string(3) " c "
}

Arg value  
Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
array(5) {
  [0]=>
  string(0) ""
  [1]=>
  string(1) "1"
  [2]=>
  string(1) " "
  [3]=>
  string(1) "a"
  [4]=>
  string(13) " 1 Array 1 c "
}

Arg value  
Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
array(5) {
  [0]=>
  string(0) ""
  [1]=>
  string(1) "1"
  [2]=>
  string(1) " "
  [3]=>
  string(1) "a"
  [4]=>
  string(13) " 1 Array 1 c "
}

Arg value  
Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
array(5) {
  [0]=>
  string(0) ""
  [1]=>
  string(1) "1"
  [2]=>
  string(1) " "
  [3]=>
  string(1) "a"
  [4]=>
  string(13) " 1 Array 1 c "
}
Error: 4096 - Object of class stdClass could not be converted to string, /Users/tjerk/Documents/php-bc/ereg/tests/split_variation_001.php(73)

Arg value  
Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
Error: 4096 - Object of class stdClass could not be converted to string, /Users/tjerk/Documents/php-bc/ereg/ereg.php(46)
array(5) {
  [0]=>
  string(0) ""
  [1]=>
  string(1) "1"
  [2]=>
  string(1) " "
  [3]=>
  string(1) "a"
  [4]=>
  string(13) " 1 Array 1 c "
}

Arg value  
Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
array(5) {
  [0]=>
  string(0) ""
  [1]=>
  string(1) "1"
  [2]=>
  string(1) " "
  [3]=>
  string(1) "a"
  [4]=>
  string(13) " 1 Array 1 c "
}

Arg value  
Error: 16384 - Function split() is deprecated, /Users/tjerk/Documents/php-bc/ereg/ereg.php(173)
array(5) {
  [0]=>
  string(0) ""
  [1]=>
  string(1) "1"
  [2]=>
  string(1) " "
  [3]=>
  string(1) "a"
  [4]=>
  string(13) " 1 Array 1 c "
}
Done